### PR TITLE
8 patch articles

### PR DESCRIPTION
**Pull request by GowerRob**
Task 8 PATCH /api/articles/:article_id

Thank you

### CODE REVIEW

File: __tests__/index.test.js [hunk 1]
```diff
@@ -193,4 +193,78 @@ describe('GET /api/articles',()=>{
...
+        return request(app)
+        .patch('/api/articles/88')
+        .send(patchData)
+        .expect(404)
+        .then((response)=>{
+               expect(response.body.msg).toBe('Article does not exist')
+        })
+    })
+
+    test('when an valid patch is posted to an invalid article id eg. pen, returns a 404 and error message',()=>{
```
Comment (RichardNeat) [inline __tests__/index.test.js]: Please update description to say 400 (be careful when copy and pasting 😉 )

File: db/controllers/controllers.js [hunk 2]
```diff
@@ -1,11 +1,12 @@
 const { fetchTopics ,
         fetchArticleById,
         fetchCommentsByArticleId,
-        fetchAllArticles} = require("../models/models");
-
-const {fetchArticleByIds} = require("../models/article.models")
+        fetchAllArticles,
+        updateArticleById} = require("../models/models");
 
+//const {fetchArticleByIds} = require("../models/article.models")
```
Comment (RichardNeat) [inline db/controllers/controllers.js]: Please remove any commented out code before submitting

File: db/controllers/controllers.js [hunk 3]
```diff
@@ -37,9 +38,9 @@ exports.getArticleById = (req, res,next)=>{
...
     const {article_id}=req.params;
 
-    const promises=[fetchCommentsByArticleId(article_id),fetchArticleByIds(article_id)]
+    const promises=[fetchCommentsByArticleId(article_id),fetchArticleById(article_id)]
     Promise.all(promises)
-    .then(([comments,articleDetails])=>{
+    .then(([comments,articleData])=>{
         res.status(200).send({comments:comments});
     }).catch((err)=>{
         next(err)
```
Comment (RichardNeat) [inline db/controllers/controllers.js:48]: There's a few changes here relating to a different endpoint. Ensure that your commits are task specific, there shouldn't be any changes to do with any other tests/implentations.

File: db/controllers/controllers.js [hunk 4]
```diff
@@ -55,3 +56,18 @@ exports.getAllArticles = (req,res,next)=>{
 }
 
 
+exports.patchArticleById =(req,res,next)=>{
+    const patchData=req.body;
+    const {article_id}=req.params;
+    const promises=[updateArticleById(patchData.inc_votes,article_id),fetchArticleById(article_id)];
```
Comment (RichardNeat) [inline db/controllers/controllers.js]: Do we need to check for the article existing before attempting to post a comment? The insert comment query will be violating a foreign key restraint if the article doesn't exist so we can rely on psql to throw an error here.

File: db/models/article.models.js [hunk 5]
```diff
@@ -1,6 +1,7 @@
 const db =require('../connection');
 
 exports.fetchArticleByIds = (article_id) =>{
+       console.log("Code 1111")
```
Comment (RichardNeat) [inline db/models/article.models.js]: Remove console.log

**Review by RichardNeat — CHANGES_REQUESTED** [review]
A few changes requested please and then resubmit. Be careful when coding on a branch, we should only have relevant code to the current task submitted for review 👍

File: db/controllers/controllers.js (hunk 4 above)
Comment (GowerRob) [reply db/controllers/controllers.js]: Okay, thanks, I have updated the code (and will address the same issue in a different task!)

File: __tests__/index.test.js [hunk 6]
```diff
@@ -193,4 +193,78 @@ describe('GET /api/articles',()=>{
...
+        .patch('/api/articles/pen')
+        .send(patchData)
+        .expect(400)
+        .then((response)=>{
+               expect(response.body.msg).toBe('Bad request')
+        })
+    })
+  
+  
+});
```
Comment (elijaharley) [inline __tests__/index.test.js:384]: Really good testing here, don't forget to test for what would happen if an empty body is passed and if a body with extra keys is passed. eg
`const patchData={inc_votes:7, extraKey: 'hello'};`

File: db/controllers/controllers.js [hunk 7]
```diff
@@ -55,3 +56,18 @@ exports.getAllArticles = (req,res,next)=>{
...
+    const patchData=req.body;
+    const {article_id}=req.params;
+    updateArticleById(patchData.inc_votes,article_id)
+    .then((article)=>{
+        res.status(201).send({article:article});
+    }).catch((err)=>{
+        next(err)
+    })
+}
+
```
Comment (elijaharley) [inline db/controllers/controllers.js:106]: Good. If we're sending back an object with the same key and value name do we need to add both?

File: db/models/models.js [hunk 8]
```diff
@@ -57,4 +56,25 @@ exports.fetchAllArticles=()=>{
...
+    .then((response)=>{
+        if (response.rowCount === 0){
+            
+            return Promise.reject({status: 404, msg: 'Article does not exist'})
+        }   
+        else {
+            return response.rows[0] 
+        }
+    });
+
```
Comment (elijaharley) [inline db/models/models.js]: Nice use of queryString here.

**Review by elijaharley — APPROVED** [review]
Good work. I've added a couple of suggestions inline for you to consider. Once you've reviewed those you're good to merge. Don't forget to .gitignore your playground files.

**Pull request merged**